proposals/0000-jsem.rst

```diff
@@ -0,0 +1,344 @@
+subprocesses spawning ``ghc`` subprocesses in turn.
+
+Effects and Interactions
+------------------------
+
+Comparison with multiple home units
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Support for multiple home units provides an alternative way to saturate the
+number of available capabilities. This is because compilation is achieved using
+a single GHC invocation, which thus doesn't have to worry about contention with
+other processes.
+
+In general, it is preferred to use multiple home units when possible, as it is
+expected to be more performant than ``-jsem``:
+
+  - no scheduling between different GHC invocations is necessary;
+  - modules are loaded directly into the home unit graph, which avoids having
+    to load the same interface files in different GHC invocations.
+
+However, it's not always possible to compile everything with a single GHC
+invocation, e.g. if the build plan involves non-Haskell dependencies somewhere
+in the middle. In comparison, the ``-jsem`` functionality can fit into any build
+system that one might be using (such as ``nix``), so it supports a wider range
+of use cases.
+
+Costs and Drawbacks
+-------------------
+
+The implementation in GHC is self-contained and doesn't impact the rest of the
+compiler much. It does however add a new flag (which interacts with ``-j``),
+and a complete implementation requires coordination with ``cabal`` and ``stack``.
+
+Alternatives
+------------
+
+GNU make supports a Jobserver protocol [9]_ [5]_ which is the same as the
+GHC Jobserver protocol described above, except that:
+
+- it uses POSIX pipes to exchange token's between processes.
+
+- participants in the protocol learn about it through environment variables
+  and the state of file descriptors on process entry.
+
+We have decided to depart in these aspects for the following reasons:
+
+- Other communities have considered and decided these aspects of the protocol
+  are unsuitable.
+
+  - ocaml [10]_ [11]_
+
+  - nix [12]_
+
+- We expect ``cabal-install`` and ``stack`` to be the only users of this feature
+  in the near term. We think the proposed protocol is adequate for this use case.
+
+- We can extend GHC to use the GNU make Jobserver protocol in the future, if
+  there are users for it.
+
+Notes:
+
+- rust's ``cargo`` does implement the GNU make Jobserver protocol [13]_ ,
+- a prototype implementation of the GNU make Jobserver protocol for GHC
+  was made by Ellie Hermaszewska [15]_ .
+
+Unresolved Questions
+--------------------
+
+Syntax of command line flags.
+
+Should we also offer configuration of this feature via environment variables?
+
+Implementation Plan
+-------------------
+
+Douglas Wilson and Sam Derbyshire have implemented a prototype at [14]_ .
+
+Once this proposal is agreed we will prepare a patch for ``cabal-install``
+and deliver it upstream.
+
+Ongoing work from Well-Typed LLP is funded by Hasura.
+
+Endorsements
+------------
+
+None at this time.
```

Comment:
I would just delete this section. In practice it is rarely used in proposals, and should be dropped IMHO.